aip/0231.md

```diff
@@ -0,0 +1,107 @@
+---
+aip:
+  id: 231
+  state: reviewing
+  created: 2019-06-18
+permalink: /231
+redirect_from:
+  - /0231
+---
+
+# Batch Get
+
+Some APIs need to allow users to get a specific set of resources at a consistent
+time point (e.g. using a read transaction). A batch get method provides this
+functionality.
+
+## Guidance
+
+APIs **may** support Batch Get using the following pattern:
+
+```proto
+rpc BatchGetBooks(BatchGetBooksRequest) returns (BatchGetBooksResponse) {
+  option (google.api.http) = {
+    get: "/v1/{parent=shelves/*}/books:batchGet"
+  };
+}
+```
+
+- The RPC's name **must** begin with `BatchGet`. The remainder of the
+  RPC name **should** be the plural form of the resource being retrieved.
+- The request and response messages **must** match the RPC name, with `-Request`
+  and `-Response` suffixes.
+- The HTTP verb **must** be `GET`.
+- The HTTP URI **must** end with `:batchGet`.
+- The URL path **should** represent the collection for the resource, matching
+  the collection used for simple CRUD operations. If the operation spans
+  parents, a dash (`-`) **may** be accepted as a wildcard.
+- For users with URL length restrictions (e.g. certain old browsers), they can
```

Comment:
I think we could deal with the confusion @alin04 points out by removing this bullet. The API producers do not actually need this information.

Reply:
Okeydokey